docs/book/src/developer/providers/v1alpha4-to-v1beta1.md

```diff
@@ -0,0 +1,17 @@
+# Cluster API v1alpha4 compared to v1beta1
+
+## Minimum Go version
+
+- The Go version used by Cluster API is still Go 1.16+
```

Comment:
I can also drop the sections where nothing changed if that's preferred. I thought it might be worth it just to state that nothing changed regarding Go / Controller-tools and kind)

Reply:
IMO we don't lose anything by being explicit here - it would be good to keep the basic format the same between versions of this text.